primitives/trie/src/lib.rs

```diff
@@ -78,6 +78,11 @@ impl<H: Hasher> TrieConfiguration for Layout<H> {
 	}
 }
 
+#[cfg(not(feature = "memory-tracker"))]
+type MemTracker = memory_db::NoopTracker<trie_db::DBValue>;
+#[cfg(feature = "memory-tracker")]
```

Comment:
I also agree with @bkchr that this solution is not ideal. This can be backfire someday with feature leaking.

Have you considered using `cfg(target_arch = "wasm32")` instead?

Reply:
I'm not a fan of introducing `#[cfg(target_arch)]` guards. They're fundamentally wrong, and will cause immense pain in the future to someone who wants to compile Substrate for a platform we didn't think about.

Reply:
That's fair, but we already using them in parity-util-mem: https://github.com/paritytech/parity-common/blob/760447f0e3791a0222ec89a4b9698f2d94a5d972/parity-util-mem/src/allocators.rs#L51

Although I agree that its API needs some love and ideally we should remove panics at runtime.

Reply:
> we already using them in parity-util-mem

Yeah, I think that this usage falls exactly into what I described:

> will cause immense pain in the future to someone who wants to compile Substrate for a platform we didn't think about